psi4/driver/task_base.py

```diff
@@ -0,0 +1,204 @@
+            mol = Molecule(**self.molecule.to_schema(dtype=2))
+
+            r = client.add_compute(
+                "psi4", self.method, self.basis, self.driver, keyword_id, [mol]
+            )
+            self.result_id = r.ids[0]
+            # NOTE: The following will re-run errored jobs by default
+            if self.result_id in r.existing:
+                ret = client.query_tasks(base_result=self.result_id)
+                if ret:
+                    if ret[0].status == "ERROR":
+                        client.modify_tasks("restart", base_result=self.result_id)
+                        logger.info("Resubmitting Errored Job {}".format(self.result_id))
+                    elif ret[0].status == "COMPLETE":
+                        logger.debug("Job already completed {}".format(self.result_id))
+                else:
+                    logger.debug("Job already completed {}".format(self.result_id))
+            else:
+                logger.debug("Submitting AtomicResult {}".format(self.result_id))
+
+            return
+
+        logger.info(f'<<< JSON launch ... {self.molecule.schoenflies_symbol()} {self.molecule.nuclear_repulsion_energy()}')
+        gof = core.get_output_file()
+
+        # EITHER ...
+        # from psi4.driver import schema_wrapper
+        # self.result = schema_wrapper.run_qcschema(self.plan())
+        # ... OR ...
+        self.result = qcng.compute(
+            self.plan(),
+            "psi4",
+            raise_error=True,
+            # local_options below suitable for serial mode where each job takes all the resources of the parent Psi4 job.
+            #   distributed runs through QCFractal will likely need a different setup.
+            local_options={
+                # B -> GiB
+                "memory": core.get_memory() / (2 ** 30),
+                "ncores": core.get_num_threads(),
+            },
+        )
+        # ... END
+
+        #pp.pprint(self.result.dict())
+        #print("... JSON returns >>>")
+        core.set_output_file(gof, True)
+        core.reopen_outfile()
+        logger.debug(pp.pformat(self.result.dict()))
+        core.print_out(_drink_filter(self.result.dict()["stdout"]))
+        self.computed = True
+
+    def get_results(self, client: Optional["FractalClient"] = None) -> AtomicResult:
+        """Return results as Atomic-flavored QCSchema."""
+
+        if self.result:
+            return self.result
+
+        if client:
+            result = client.query_results(id=self.result_id)
+            logger.debug(f"Querying AtomicResult {self.result_id}")
+            if len(result) == 0:
+                return self.result
+
+            self.result = result[0]
+            return self.result
+
+
+def _drink_filter(stdout: str) -> str:
+    """Don't mess up the widespread ``grep beer`` test of Psi4 doneness by printing multiple drinks per outfile."""
+
+    stdout = stdout.replace("\n*** Psi4 exiting successfully. Buy a developer a beer!", "")
```

Comment:
Why does this not create a problem by removing the very last one?

Reply:
Because it's only removing drinks from the task outputs. There's still a drink when the governing psi4 session completes:

```
  //>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>//
  //      FiniteDifference Results     //
  //<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<//

  Computing second-derivative from gradients using projected,
  symmetry-adapted, cartesian coordinates.

  9 gradients passed in, including the reference geometry.
  Generating complete list of displacements from unique ones.

    Operation 3 takes plus displacements of irrep A2 to minus ones.
    Operation 2 takes plus displacements of irrep B1 to minus ones.
    Operation 2 takes plus displacements of irrep B2 to minus ones.

-------------------------------------------------------------
    SCF hessian by gradient...............................................................PASSED

    Psi4 stopped on: Tuesday, 17 May 2022 11:15AM
    Psi4 wall time for execution: 0:00:11.25

*** Psi4 exiting successfully. Buy a developer a beer!
```